File: src/codes/scattered-instaniation/php/uncode.php
<?php
class Logger {
  public function log(string $message): void {
    echo "[LOG]: {$message}\n";
  }
}

class UserService {
  private Logger $logger;

  public function __construct() {
    $this->logger = new Logger(); // 毎回 new!!
  }

  public function createUser(string $name): void {
    $this->logger->log("ユーザー作成: {$name}");
  }
}

File: src/codes/scattered-instaniation/py/uncode.py
class Logger:
    def log(self, message: str):
        print(f"[LOG]: {message}")

class UserService:
    def __init__(self):
        self.logger = Logger()  # 毎回 new!!

    def create_user(self, name: str):
        self.logger.log(f"ユーザー作成: {name}")
